nexus-repository-conan/src/main/java/org/sonatype/repository/conan/upgrade/ConanUpgrade_1_1.java

```diff
@@ -0,0 +1,171 @@
+      List<String> repositoryNames = findConanRepositoryNames();
+      updateAssetPath(repositoryNames);
+      removeAttributesFromConanManifest(repositoryNames);
+    }
+  }
+
+  private List<String> findConanRepositoryNames() {
+    try (ODatabaseDocumentTx oDatabaseDocumentTx = configDatabaseInstance.get().connect()) {
+      final List<ODocument> documents = oDatabaseDocumentTx.query(
+          new OSQLSynchQuery<ODocument>("select from repository where recipe_name in ['conan-proxy', 'conan-hosted']"));
+      return documents
+          .stream()
+          .map(entries -> (String) entries.field(P_REPOSITORY_NAME))
+          .collect(Collectors.toList());
+    }
+  }
+
+  private void removeAttributesFromConanManifest(final List<String> repositoryNames) {
+    DatabaseUpgradeSupport.withDatabaseAndClass(componentDatabaseInstance, ASSET_CLASS_NAME,
+        (db, type) -> findAssets(db, repositoryNames,
+            "select from asset where bucket = ? and attributes.conan.asset_kind = 'CONAN_MANIFEST'")
+            .forEach(oDocument -> {
+
+              Map<String, Object> attributes = oDocument.field(P_ATTRIBUTES);
+              // remove all attributes, except asset_kind from conan "bucket"
+              attributes
+                  .put(ConanFormat.NAME, Collections.singletonMap(P_ASSET_KIND, AssetKind.CONAN_MANIFEST.name()));
+              oDocument.field(P_ATTRIBUTES, attributes);
+
+              oDocument.save();
+            })
+    );
+  }
+
+  private void updateAssetPath(final List<String> repositoryNames) {
+    DatabaseUpgradeSupport.withDatabaseAndClass(componentDatabaseInstance, ASSET_CLASS_NAME,
+        (db, type) -> findAssets(db, repositoryNames, "select from asset where bucket = ?")
+            .forEach(oDocument -> {
+              String name = oDocument.field(P_ASSET_NAME);
+              String nextName = null;
+
+              String strategy1 = "/v1/conans/v1/conans/";
+              String strategy2 = "v1/conans/";
+              String strategy3 = "/v1/conans/";
+              String conans = "conans/";
+
+              if (name.startsWith(strategy1)) { // broken based on refactor-v1-api HOSTED
+                nextName = conans + name.substring(strategy1.length());
+              }
+              else if (name.startsWith(strategy2)) { // based on refactor-v1-api PROXY
+                nextName = conans + name.substring(strategy2.length());
+              }
+              else if (name.startsWith(strategy3)) { // latest(master) HOSTED
+                nextName = conans + name.substring(strategy3.length());
+              }
+              else if (!name.startsWith(conans)) { // latest(master) PROXY
+                nextName = conans + name;
+              }
+
+              if (nextName != null) {
+                oDocument.field(P_ASSET_NAME, nextName);
+                oDocument.save();
+              }
+            })
+    );
+  }
+
+  private Stream<ODocument> findAssets(
```

Comment:
Is it necessary to return stream here, List will be clearer up to me?

Reply:
I need foreach.
Yes, you are right, I can find assets and after use collect(Collectors.toList()) and use List
But, I need class which has foreach, so I can skip transformation collect(Collectors.toList()) and just use Stream